src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpMetadataContext.cs

```diff
@@ -9,27 +9,38 @@ namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator
     internal struct CSharpMetadataContext
     {
         internal readonly ImmutableArray<MetadataBlock> MetadataBlocks;
+        internal readonly ImmutableArray<Alias> Aliases;
         internal readonly CSharpCompilation Compilation;
         internal readonly EvaluationContext EvaluationContext;
 
-        internal CSharpMetadataContext(ImmutableArray<MetadataBlock> metadataBlocks, CSharpCompilation compilation)
+        internal CSharpMetadataContext(
+            ImmutableArray<MetadataBlock> metadataBlocks, 
+            ImmutableArray<Alias> aliases,
+            CSharpCompilation compilation)
         {
             this.MetadataBlocks = metadataBlocks;
+            this.Aliases = aliases;
             this.Compilation = compilation;
             this.EvaluationContext = null;
         }
 
-        internal CSharpMetadataContext(ImmutableArray<MetadataBlock> metadataBlocks, EvaluationContext evaluationContext)
+        internal CSharpMetadataContext(
+            ImmutableArray<MetadataBlock> metadataBlocks,
+            ImmutableArray<Alias> aliases, 
+            EvaluationContext evaluationContext)
         {
             this.MetadataBlocks = metadataBlocks;
+            this.Aliases = aliases;
             this.Compilation = evaluationContext.Compilation;
             this.EvaluationContext = evaluationContext;
         }
 
-        internal bool Matches(ImmutableArray<MetadataBlock> metadataBlocks)
+        internal bool Matches(ImmutableArray<MetadataBlock> metadataBlocks, ImmutableArray<Alias> aliases)
         {
             return !this.MetadataBlocks.IsDefault &&
-                this.MetadataBlocks.SequenceEqual(metadataBlocks);
+                this.MetadataBlocks.SequenceEqual(metadataBlocks) &&
+                !this.Aliases.IsDefault &&
```

Comment:
Is this.Aliases.IsDefault true only in the default(CSharpMetadataContext) case? If so, this check is not necessary since !this.MetadataBlocks.IsDefault would be false.